### DISCUSSION TITLE
Directus images disappear from media library on Heroku

### DISCUSSION THREAD

**Discussion by GreetVdL** [directus/directus#13854] [Archive]
Hi. I created a Next.js project that is deployed on Vercel and uses a MySQL database. I then deployed a Directus instance on Heroku that is tied to that same database. In my Next.js project I want to fetch and render images that I uploaded to Directus. At first this works, but after a while all the images disappear in the Directus media library. The folders and references to the images are still there, but I don't see the pictures anymore, I get to see a JPG logo instead. When I try to fetch the images I get a 502 "Bad Gateway" error. I don't know what causes the images to disappear and how to fix this.
![Directus-images-missing](https://user-images.githubusercontent.com/43548163/173238863-20a35384-154f-4892-9772-47fd5ef82851.png)

**Comment by rijkvanzanten** [comment]
Heya! Heroku uses an ephemeral filesystem meaning that all files will be deleted when the dyno restarts. https://devcenter.heroku.com/articles/dynos#ephemeral-filesystem. You'll have to use some persistent file storage service, like AWS S3, to properly manage files when hosting on Heroku 👍🏻

**Comment by GreetVdL** [comment]
OK, thank you very much for your answer! I am going to try to solve this issue with an Heroku addon then.
